comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package process implements the local process collector for
+// Workloadmeta.
+package process
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"go.uber.org/fx"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID       = "local-process-collector"
+	cacheValidityNoRT = 2 * time.Second
+	componentName     = "workloadmeta-process"
+)
+
+type collector struct {
+	id      string
+	store   workloadmeta.Component
+	catalog workloadmeta.AgentType
+
+	processDiffCh <-chan *processwlm.ProcessCacheDiff
+}
+
+// NewCollector returns a new docker collector provider and an error
+func NewCollector() (workloadmeta.CollectorProvider, error) {
+	return workloadmeta.CollectorProvider{
+		Collector: &collector{
+			id:      collectorID,
+			catalog: workloadmeta.NodeAgent,
+		},
+	}, nil
+}
+
+// GetFxOptions returns the FX framework options for the collector
+func GetFxOptions() fx.Option {
+	return fx.Provide(NewCollector)
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Component) error {
+	if !config.Datadog().GetBool("language_detection.enabled") && flavor.GetFlavor() != flavor.DefaultAgent {
+		return errors.NewDisabled(componentName, "core agent language detection not enabled")
+	}
+	c.store = store
+	c.processDiffCh = processwlm.GetSharedWorkloadMetaExtractor(config.SystemProbe).ProcessCacheDiff()
+
+	go c.stream(ctx)
+
+	return nil
+}
+
+func CollectorEventsFromProcessDiff(creations []*processwlm.ProcessEntity, deletions []*processwlm.ProcessEntity) []workloadmeta.CollectorEvent {
```

Comment:
Wouldn't it be simpler to pass the diff itself as an argument here? 

I think it would be more aligned with the function name. 

I don't have a strong opinion though, both should work fine.

Reply:
Done